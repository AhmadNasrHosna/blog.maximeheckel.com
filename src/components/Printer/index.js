import React from 'react';
import styled from '@emotion/styled';

const Wrapper = styled('div')`
  width: 800px;
  height: 400px;
  color: #2b2d3e;
  font-family: 'Helvetica', sans-serif;
  display: flex;
  background: #fff;
  svg {
    max-width: 200px;
    min-width: 200px;
  }
`;

const Title = styled('h1')`
  padding-left: 20px;
  margin-top: 30px;
  margin-bottom: 20px;
  font-size: 38px;
  font-weight: 500;
  line-height: 1.5;
`;

const LocalLogo = () => (
  <svg
    width="401"
    height="401"
    viewBox="0 0 401 401"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <circle cx="200.633" cy="200.58" r="200" fill="none" />
      <rect
        x="131.283"
        y="104.174"
        width="46.5558"
        height="144.938"
        rx="23.2779"
        transform="rotate(26 131.283 104.174)"
        fill="#fff"
      />
      <rect
        x="231.663"
        y="109.137"
        width="46.5558"
        height="144.938"
        rx="23.2779"
        transform="rotate(26 231.663 109.137)"
        fill="#fff"
      />
      <rect
        x="257.779"
        y="207.753"
        width="46.5558"
        height="68.1965"
        rx="23.2779"
        transform="rotate(-30 257.779 207.753)"
        fill="#fff"
      />
    </g>
  </svg>
);

const PrinterComponent = ({ title }) => (
  <Wrapper>
    <div
      style={{ backgroundColor: '#2b2d3e', maxWidth: '250px', height: '100%' }}
    >
      <LocalLogo />
    </div>
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
      }}
    >
      <Title>{title}</Title>
      <h3 style={{ color: '#196FD8', paddingLeft: '20px' }}>@MaximeHeckel</h3>
    </div>
  </Wrapper>
);

export default PrinterComponent;
