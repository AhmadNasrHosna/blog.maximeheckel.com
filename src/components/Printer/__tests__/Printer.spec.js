import 'jest-dom/extend-expect';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Printer from '../';

afterEach(() => {
  cleanup();
});

describe('Printer', () => {
  it('Renders the Printer component with the proper title', () => {
    const { asFragment, container } = render(<Printer title="my post" />);
    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('h1')).toHaveTextContent('my post');
  });
});
