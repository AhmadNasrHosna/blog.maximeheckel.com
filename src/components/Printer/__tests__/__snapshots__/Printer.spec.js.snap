// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Printer Renders the Printer component with the proper title 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 800px;
  height: 400px;
  color: #2b2d3e;
  font-family: 'Helvetica',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fff;
}

.emotion-1 svg {
  max-width: 200px;
  min-width: 200px;
}

.emotion-0 {
  padding-left: 20px;
  margin-top: 30px;
  margin-bottom: 20px;
  font-size: 38px;
  font-weight: 500;
  line-height: 1.5;
}

<div
    class="emotion-1"
  >
    <div
      style="background-color: rgb(43, 45, 62); max-width: 250px; height: 100%;"
    >
      <svg
        fill="none"
        height="401"
        viewBox="0 0 401 401"
        width="401"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <circle
            cx="200.633"
            cy="200.58"
            fill="none"
            r="200"
          />
          <rect
            fill="#fff"
            height="144.938"
            rx="23.2779"
            transform="rotate(26 131.283 104.174)"
            width="46.5558"
            x="131.283"
            y="104.174"
          />
          <rect
            fill="#fff"
            height="144.938"
            rx="23.2779"
            transform="rotate(26 231.663 109.137)"
            width="46.5558"
            x="231.663"
            y="109.137"
          />
          <rect
            fill="#fff"
            height="68.1965"
            rx="23.2779"
            transform="rotate(-30 257.779 207.753)"
            width="46.5558"
            x="257.779"
            y="207.753"
          />
        </g>
      </svg>
    </div>
    <div
      style="display: flex; flex-direction: column; justify-content: center;"
    >
      <h1
        class="emotion-0"
      >
        my post
      </h1>
      <h3
        style="color: rgb(25, 111, 216); padding-left: 20px;"
      >
        @MaximeHeckel
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
